### PR TITLE
Fh 3123 sync horizontal scaling

### PR DESCRIPTION
**Pull request by wei-lee**
Add support for scaling sync horizontally

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,18 +29,20 @@ npm install --save fh-mbaas-api
 
 This will install the latest version of fh-mbaas-api and save the installed version in your package.json
```
Comment (evanshortiss) [inline README.md]: @wei-lee should we add the required lines that should be added to express apps?

Comment (wei-lee) [inline README.md]: @evanshortiss not sure what do u mean by this? Do you mean the code to init the new sync?

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,43 @@
+node('uart-and-cloud-deploy-slave') {
```
Comment (david-martin) [inline Jenkinsfile]: Should we remove this file as it isn't being used?

File: lib/sync/worker.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+function initBackoff(backOffOpts) {
+  var bo;
+  if (backOffOpts.strategy && backOffOpts.strategy.toLowerCase() === 'exp') {
+    bo = new backoff.ExponentialStrategy({initialDelay: backOffOpts.min, maxDelay: backOffOpts.max});
+  } else if (backOffOpts.strategy &&  backOffOpts.strategy.toLowerCase() === 'fib') {
+    bo = new backoff.FibonacciStrategy({initialDelay: backOffOpts.min, maxDelay: backOffOpts.max});
+  } else {
+    //if no valid value is given, just return the default minimum delay. Can be used to turn off backoff.
+    console.log('[warning] no valid strategy found in the backoff options %o. Turning off backoff', backOffOpts);
```
Comment (david-martin) [inline lib/sync/worker.js]: Should this include 'sync' somewhere in the log?
e.g. 'No valid sync strategy found in the backoff options %o. Turning off sync worker backoff'

File: package.json [hunk 4]
```diff
@@ -1,30 +1,36 @@
 {
   "name": "fh-mbaas-api",
-  "version": "6.1.8",
+  "version": "7.0.0-0",
```
Comment (david-martin) [inline package.json]: @wei-lee Lets set this to 7.0.0 for merging to master

File: Jenkinsfile (hunk 2 above)
Comment (wei-lee) [inline Jenkinsfile]: sure

File: lib/sync/worker.js (hunk 3 above)
Comment (wei-lee) [inline lib/sync/worker.js]: ok

File: package.json (hunk 4 above)
Comment (wei-lee) [inline package.json]: will do

**Pull request merged**